### DISCUSSION TITLE
怎么将PaoPaoDNS作为Android手机的自定义DNS服务器?

### DISCUSSION THREAD

**Discussion by biggbuddy** [kkkgo/PaoPaoDNS#225] [问答和帮助]
我的HomeLab是 主路由 OpenWrt + PaoPaoDNS + PaoPaoGateway 的结构. 
开了一个域名  dns.example.com 在主路由上开了相关的端口, 把主路由的53端口遇到到了 PaoPaoDNS 的53端口, 已经在itdog上验证域名和端口是通的. 但把  dns.example.com 配置到android手机的自定义DNS上的时候, 提示无法连接. 

问了ChatGPT, 说是android自定义DNS服务器使用的是  DoT, 需要监听853端口, 所以把主路由对外 开放的端口改为853, 协议改为TCP, 映射的PaoPaoDNS端口为5304, 在itdog上验证端口也是通的, 但Android手机上还是提示无法连接. 

![image](https://github.com/user-attachments/assets/562df8b4-5f0f-44ca-bcac-b48fc6046f1c)

PaoPaoDNS的各个端口都尝试过了, 结果都是无法连接, 不知道大家有碰到过这样的问题吗?

**Comment by kkkgo** [comment]
你对DoT 的理解有误。
DoT 全称是 DNS over TLS，它使用 TLS 加密协议来传输 DNS 协议，并不是简单的转发端口到普通DNS服务。

**Reply by biggbuddy** [reply]
好吧. 

可否大概说一下在OpenWrt作为主路由拨号, 能获得IPv6公网IP, PaoPaoDNS在另一个虚拟机结点的结构下要实现Android自定义DNS的思路?

**Reply by kkkgo** [reply]
你要搭建DoH/DoT的话，可以搜一下Adguadhome的相关教程，网页界面直接操作。
